### PR TITLE
layers: add image layout validation to vkCmdBlitImage()

### PR DESCRIPTION
**Pull request by NorbertGarnysAMD**

### CODE REVIEW

File: layers/buffer_validation.cpp [hunk 1]
```diff
@@ -2423,8 +2424,15 @@ bool PreCallValidateCmdBlitImage(layer_data *device_data, GLOBAL_CB_NODE *cb_nod
         }  // Depth or Stencil
 
         // Do per-region checks
-        for (uint32_t i = 0; i < regionCount; i++) {
-            VkImageBlit rgn = pRegions[i];
+        for (uint32_t i = 0; i < region_count; i++) {
+            VkImageBlit rgn = regions[i];
+            bool hit_error = false;
+            skip |=
+                VerifyImageLayout(device_data, cb_node, src_image_state, regions[i].srcSubresource, src_image_layout,
```
Comment (tobine) [inline layers/buffer_validation.cpp]: Could use rgn here instead of regions[i].

File: layers/buffer_validation.cpp [hunk 2]
```diff
@@ -2423,8 +2424,15 @@ bool PreCallValidateCmdBlitImage(layer_data *device_data, GLOBAL_CB_NODE *cb_nod
...
+                                  VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL, "vkCmdBlitImage()", VALIDATION_ERROR_184001bc, &hit_error);
+            skip |=
+                VerifyImageLayout(device_data, cb_node, dst_image_state, regions[i].dstSubresource, dst_image_layout,
```
Comment (tobine) [inline layers/buffer_validation.cpp]: Could use rgn here instead of regions[i].

File: layers/buffer_validation.cpp [hunk 3]
```diff
@@ -2423,8 +2424,15 @@ bool PreCallValidateCmdBlitImage(layer_data *device_data, GLOBAL_CB_NODE *cb_nod
         }  // Depth or Stencil
 
         // Do per-region checks
-        for (uint32_t i = 0; i < regionCount; i++) {
-            VkImageBlit rgn = pRegions[i];
+        for (uint32_t i = 0; i < region_count; i++) {
+            VkImageBlit rgn = regions[i];
```
Comment (tobine) [inline layers/buffer_validation.cpp]: const?

File: tests/layer_validation_tests.cpp [hunk 4]
```diff
@@ -3242,6 +3257,168 @@ TEST_F(VkLayerTest, BlitImageFilters) {
...
+    VkResult err;
+    VkFormat fmt = VK_FORMAT_R8G8B8A8_UNORM;
+
+    VkSubmitInfo submit_info = {};
+    submit_info.sType = VK_STRUCTURE_TYPE_SUBMIT_INFO;
+    submit_info.commandBufferCount = 1;
+    submit_info.pCommandBuffers = &m_commandBuffer->handle();
+
+    // Create images
+    VkImageObj imgSrcTransfer(m_device);
```
Comment (tobine) [inline tests/layer_validation_tests.cpp]: kill camelCase on these var names

File: tests/layer_validation_tests.cpp [hunk 5]
```diff
@@ -3242,6 +3257,168 @@ TEST_F(VkLayerTest, BlitImageFilters) {
...
+    ASSERT_TRUE(imgDstTransfer.initialized());
+    ASSERT_TRUE(imgGeneral.initialized());
+    ASSERT_TRUE(imgColor.initialized());
+
+    imgSrcTransfer.SetLayout(VK_IMAGE_ASPECT_COLOR_BIT, VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL);
+    imgDstTransfer.SetLayout(VK_IMAGE_ASPECT_COLOR_BIT, VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL);
+    imgGeneral.SetLayout(VK_IMAGE_ASPECT_COLOR_BIT, VK_IMAGE_LAYOUT_GENERAL);
+    imgColor.SetLayout(VK_IMAGE_ASPECT_COLOR_BIT, VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL);
+
+    VkImageBlit blitRegion = {};
```
Comment (tobine) [inline tests/layer_validation_tests.cpp]: camelCase

**Comment by NorbertGarnysAMD** [comment]
Thanks for review. I've changed the code according to your comments and split it to 2 commits.

**Pull request merged**